src/core_modules/capture-core/components/D2Form/D2Form.component.js

```diff
@@ -135,21 +121,41 @@ class D2Form extends React.PureComponent<Props> {
     )
 
     render() {
-        const { formFoundation, id, classes, ...passOnProps } = this.props;
-        const formHorizontal = this.props.formHorizontal;
+        const {
+            formFoundation,
+            id,
+            classes,
+            isFormInReduxStore,
+            formHorizontal,
+            ...passOnProps
+        } = this.props;
         const metaDataSectionsAsArray = Array.from(formFoundation.sections.entries()).map(entry => entry[1]);
 
         const sections = metaDataSectionsAsArray.map(section => (formHorizontal ? this.renderHorizontal(section, passOnProps) : this.renderVertical(section, passOnProps, classes)));
 
         return (
             <React.Fragment>
-                {sections}
+                {
+                    isFormInReduxStore
+                        ?
+                        sections
+                        :
+                        log.error(
```

Comment:
the component will throw an error when the user wont add the form id into the redux store. this way we let the user know of what they need to do to use this component in the future